transaction-error/src/lib.rs

```diff
@@ -42,9 +42,12 @@ pub enum TransactionError {
     /// the `recent_blockhash` has been discarded.
     BlockhashNotFound,
 
-    /// An error occurred while processing an instruction. The first element of the tuple
-    /// indicates the instruction index in which the error occurred.
-    InstructionError(u8, InstructionError),
+    /// An error occurred while processing an instruction. The first element of the tuple indicates
+    /// the index of the outer instruction in which the error occurred, and the third the account
+    /// index of the program responsible for the error (ie. the error may have originated from an
+    /// inner instruction). The account index of the responsible program may be `None` for
```

Comment:
I'm not sure this approach actually clears up much ambiguity. There could be multiple inner instructions that invoke the same program. In order to pinpoint the failing inner instruction you would still need to rely on the inner ix metadata that we already provide.

Reply:
Good catch! Though the [original goal](https://github.com/anza-xyz/kit/issues/149) was to be able to correlate what program `Custom(###)` pertains to, we could _also_ add the path to the actual inner instruction.

Reply:
Sorry I don't really follow your response. Is the "original goal" still the current goal? When you say "we could also add the path to the actual inner instruction" is that rhetorical? If this is a "good catch," does this warrant any change with your current approach?

Reply:
Is it correct to say that you're just trying to map a custom program error code to the actual program that returned it?

Reply:
If we're just mapping custom error codes, do we need to add the tx account index for the failing program to all instruction error variants?

Reply:
> If we're just mapping custom error codes…

We're not. That was the original impetus for this change, but the exploration led to the insight that _most_ `InstructionErrors` would benefit from know what program caused them.

Reply:
> If this is a "good catch," does this warrant any change with your current approach?

Yep! It warrants me adding the inner instruction index so that you can – for instance – tell that a `InsufficientFundsForRent` came from instruction 1.4 as opposed to 1.2, even though 1.2 and 1.4 have the same program address.

Reply:
I added the index of the inner instruction here in the type, and [here](https://github.com/anza-xyz/agave/pull/6083) in the implementation, @jstarry!